src/test/java/reactor/netty/http/client/HttpRedirectTest.java

```diff
@@ -583,4 +590,57 @@ public void testLastLocationSetToResourceUrlOnRedirect() throws CertificateExcep
 		initialServer.disposeNow();
 		redirectServer.disposeNow();
 	}
+
+	@Test
+	public void testBuffersForRedirectWithContentShouldBeReleased() {
+		doTestBuffersForRedirectWithContentShouldBeReleased("Redirect response content!");
+	}
+
+	@Test
+	public void testBuffersForRedirectWithLargeContentShouldBeReleased() {
+		doTestBuffersForRedirectWithContentShouldBeReleased(StringUtils.repeat("a", 10000));
+	}
+
+	private void doTestBuffersForRedirectWithContentShouldBeReleased(String redirectResponseContent) {
+		final String initialPath = "/initial";
+		final String redirectPath = "/redirect";
+
+		DisposableServer server =
+				HttpServer.create()
+						.port(0)
+						.route(r -> r
+								.get(initialPath, (req, res) -> res
+										.status(HttpResponseStatus.MOVED_PERMANENTLY)
+										.header(HttpHeaderNames.LOCATION, redirectPath)
+										.sendString(Mono.just(redirectResponseContent)))
```

Comment:
What will happen with the connection if you send `Flux.just(redirectResponseContent, redirectResponseContent)`?

Reply:
I think we have issue with the connection staying active when we do a redirection.

Reply:
Should I create a issue for the connection staying active so that we do not hijact this PR?

Reply:
Yes, I think that would be the best. Thank you.

Reply:
#1027